src/Core/Migration/MigrationHandler.php

```diff
@@ -229,9 +237,42 @@ protected function buildMigrationQueries()
         }
 
         $oDirectory = new RecursiveDirectoryIterator($this->migrationQueriesDir);
-        $oFlattened = new RecursiveIteratorIterator($oDirectory);
+        $oIterators = new AppendIterator();
+        $oIterators->append(new RecursiveIteratorIterator($oDirectory));
+
+        // Include module migration files
+        if (!self::$skipModuleMigration) {
+            $oConfig = Registry::getConfig();
+
+            if (!class_exists(ModuleList::class)) {
+                print "ERROR: Oxid ModuleList class can not be loaded,
+                please try to run vendor/bin/oe-eshop-unified_namespace_generator";
+            } else {
+                try {
+                    // TODO: We need to use shop internal service to get active modules path
+                    $moduleList = oxNew(ModuleList::class);
+                    $modulesDir = $oConfig->getModulesDir();
+                    $activeModules = $moduleList->getActiveModuleInfo();
+
+                    if (is_array($activeModules) and count($activeModules) > 0) {
+                        foreach ($activeModules as $activeModule) {
+                            $migrationQueryDir = $modulesDir . $activeModule . DIRECTORY_SEPARATOR .
+                                'migration' . DIRECTORY_SEPARATOR;
+                            if (!is_dir($migrationQueryDir)) {
+                                continue;
+                            }
+                            $oDirectory = new RecursiveDirectoryIterator($migrationQueryDir);
+                            $oIterators->append(new RecursiveIteratorIterator($oDirectory));
+                        }
+                    }
+                } catch (Throwable $exception) {
+                    print "Shop is not able to list modules\n";
+                    print $exception->getMessage();
```

Comment:
Just throw an exception here, as we do it in https://github.com/OXIDprojects/oxid-console/blob/ab98dbfbe22fa5a6b63c4dfc0d7a0e36a39938c6/src/Core/Migration/MigrationHandler.php#L70

If really needed you can add additional Infos to the exception and re-throw it. If the exception contains already all needed data just remove the `try-catch` wrapper.

Reply:
Adapted the changes as per suggestion. 
I think below code block is good here.